server/src/main/java/edu/colostate/cs/cs414/chesshireCoders/jungleServer/handlers/LoginHandler.java

```diff
@@ -0,0 +1,80 @@
+package edu.colostate.cs.cs414.chesshireCoders.jungleServer.handlers;
+
+import com.esotericsoftware.kryonet.Listener;
+
+import edu.colostate.cs.cs414.chesshireCoders.jungleServer.dataAccessObjects.LoginDAO;
+import edu.colostate.cs.cs414.chesshireCoders.jungleServer.server.JungleDB;
+import edu.colostate.cs.cs414.chesshireCoders.jungleUtil.game.Login;
+import edu.colostate.cs.cs414.chesshireCoders.jungleUtil.requests.LoginRequest;
+import edu.colostate.cs.cs414.chesshireCoders.jungleUtil.responses.LoginResponse;
+import edu.colostate.cs.cs414.chesshireCoders.jungleUtil.responses.ResponseStatusCodes;
+
+import edu.colostate.cs.cs414.chesshireCoders.jungleUtil.security.Crypto;
+
+import com.esotericsoftware.kryonet.Connection;
+
+import java.sql.SQLException;
+
+public class LoginHandler extends Listener{
+	
+	private JungleDB jungleDB = JungleDB.getInstance();
+	
+	@Override
+    public void received(Connection connection, Object received) {
+
+        if (received instanceof LoginRequest) {
+            try {
+                connection.sendTCP(handleLogin((LoginRequest) received));
+            } catch (SQLException e) {
+                connection.sendTCP(new LoginResponse(
+                        ResponseStatusCodes.SERVER_ERROR,
+                        e.getMessage()
+                ));
+            }
+        } 
+    }
+	
+	private LoginResponse handleLogin(LoginRequest request) throws SQLException {
+		
+		java.sql.Connection connection = jungleDB.getConnection();
+
+        LoginDAO loginDAO = new LoginDAO(connection);
+		
+		
+        try {
+            connection.setAutoCommit(false);
```

Comment:
Turning off auto-commit isn't necessary here since we should only be retrieving data from the database.

We shouldn't be adding any new data. And turning this off is only helpful when we have multiple database updates/inserts/transactions that we want to be able to rollback or commit as one unit.

Reply:
more info about DB transactions [here](http://www.tutorialspoint.com/sql/sql-transactions.htm)

Reply:
An example of when you would want to turn off auto-commit would be when you are adding a new user to the database.

When adding a new user, A new row in the `User` table is created, _and_ a new row in `Login` is created. But what happens when the insertion into User succeeds, but the insertion into Login fails (maybe the email is already taken)?

We don't want to keep the new row in user if adding the login info failed. So we turn off auto-committing of updates to the database. Then if something goes wrong, we can call `rollback`, and any changes we made will be reverted. 

And if all goes well, we can call `commit` to tell the database that we are happy with our changes, and they should be made permanent.

Reply:
That makes sense. I removed any setting of AutoCommit from the code.